src/utils/portfolioCalculations/getCurrentPortfolio.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  getSortedPurchaseTransactions,
+  getSortedSaleTransactions,
+} from './getSortedTransactions';
+import {
+  calculationParams,
+  CurrentPortfolioItem,
+  CurrentPortfolioTransaction,
+} from '../../types/calculations';
+
+export const getCurrentPortfolio = ({
+  existingTransactions,
+  savedTransactions,
+}: calculationParams) => {
+  const transactions = [...existingTransactions, ...savedTransactions];
+
+  const purchaseTransactions: CurrentPortfolioTransaction[] =
+    getSortedPurchaseTransactions(transactions).map((tx) => ({
+      ...tx,
+      remainingNumberOfStocksOwned: tx.numberOfStocks,
+    }));
+  const saleTransactions = getSortedSaleTransactions(transactions).map(
+    (tx) => ({
+      ...tx,
+      remainingNumberOfStocksToSell: tx.numberOfStocks,
+    }),
+  );
+
+  saleTransactions.forEach((saleTx) => {
+    let remainingNumberOfStocksToSell = saleTx.remainingNumberOfStocksToSell;
+
+    purchaseTransactions.forEach((purchaseTx) => {
+      if (remainingNumberOfStocksToSell <= 0) return;
+      if (purchaseTx.holding.holdingSymbol !== saleTx.holding.holdingSymbol)
+        return;
+
+      const stocksToSell = Math.min(
+        remainingNumberOfStocksToSell,
+        purchaseTx.remainingNumberOfStocksOwned,
+      );
+
+      purchaseTx.remainingNumberOfStocksOwned -= stocksToSell;
+      remainingNumberOfStocksToSell -= stocksToSell;
+    });
+  });
+
+  const currentPortfolio = purchaseTransactions.filter(
+    (tx) => tx.remainingNumberOfStocksOwned > 0,
+  );
+
+  const formattedPortfolio = formatCurrentPortfolio(currentPortfolio);
+
+  return formattedPortfolio;
+};
+
+const formatCurrentPortfolio = (
```

Comment:
A function should be defined before it's used. Try to search for "const vs function" to learn the differences :) 
Ideally this function can be imported as well.